medusa/server/api/v1/core.py

```diff
@@ -145,16 +149,17 @@ def _out_as_image(self, _dict):
         return _dict['image'].get_media()
 
     def _out_as_json(self, _dict):
-        self.set_header("Content-Type", "application/json;charset=UTF-8")
+        self.set_header('Content-Type', 'application/json;charset=UTF-8')
         try:
             out = json.dumps(_dict, ensure_ascii=False, sort_keys=True)
             callback = self.get_query_argument('callback', None) or self.get_query_argument('jsonp', None)
             if callback:
-                out = callback + '(' + out + ');'  # wrap with JSONP call if requested
-        except Exception as e:  # if we fail to generate the output fake an error
-            logger.log(u"API :: " + traceback.format_exc(), logger.DEBUG)
-            out = '{"result": "%s", "message": "error while composing output: %s"}' % \
-                  (result_type_map[RESULT_ERROR], ex(e))
+                out = '{0}({1});'.format(callback, out)  # wrap with JSONP call if requested
+        except Exception as error:  # if we fail to generate the output fake an error
+            log.exception(u'API :: Traceback')
+            out = '{{"result": "{0}", "message": "error while composing output: {1!r}"}}'.format(
```

Comment:
Because I'm using format now instead of %s. Format uses {} as special chars. So you have to escape them.